multidict/_pair_list.c

```diff
@@ -1115,13 +1115,66 @@ pair_list_update_from_seq(PyObject *op, PyObject *seq)
     return -1;
 }
 
-PyObject *
-_pair_list_calc_identity(PyObject *op, PyObject *key)
+int
+pair_list_eq_to_mapping(PyObject *op, PyObject *other)
 {
-    pair_list_t *list = (pair_list_t *)op;
-    return list->calc_identity(key);
+    PyObject *key = NULL;
+    PyObject *avalue = NULL;
+    PyObject *bvalue = NULL;
+
+    Py_ssize_t pos;
+    
+    int cmp;
+    int is_eq;
+
+    if (!PyMapping_Check(other)) {
+        PyErr_Format(PyExc_TypeError,
+                     "other argument must be a mapping, not %s",
+                     Py_TYPE(other)->tp_name);
+        return -1;
+    }
+
+    if (pair_list_len(op) != PyMapping_Length(other)) {
+        return 0;
+    }
+
+    pos = 0;
+    is_eq = 1;
+    while (pair_list_next(op, &pos, NULL, &key, &avalue)) {
+        if (!PyMapping_HasKey(other, key)) {
```

Comment:
The check is redundant.
`PyObject_GetItem` should be enough.
If its return value is `NULL` call `PyErr_Clear()` and assign `is_eq` to `0`.
Comparison to mapping is not the hot path (no need a specialization for dict instance) but I see no reasons to lookup a mapping twice.

Reply:
Yes, you right.